### ISSUE TITLE
Upgrade from 1.7 to 1.8 on a Non NAT AGW fails

### ISSUE THREAD

**Issue by Sujay-Radtonics** [magma/magma#14826]
### Your Environment

- **Version:** - 1.7, 1.8
- **Affected Component:** Access Gateway
- **Affected Subcomponent:**  AGW service
- **Deployment Environment:** AGW (baremetal or VM)

### Describe the Issue

Upgrade from release 1.7 to 1.8 on a Non NAT AGW fails. It stops abruptly and partial upgrade happens

#### To Reproduce

1. Have AGW in Non NAT mode in release 1.7
2. run upgrade script to upgrade to 1.8.
3. Upgrade process starts and while the ovs upgrade is in progress, ssh session gets stuck. 

#### Expected behavior

Upgrade should happen without any errors

#### Additional context

Observation is that while OVS upgrade is in progress, uplink_br0 looses its IP address. There by access to the AGW is lost. 
Upgrade is partial.

**Comment by nstng** [comment]
Are you able to connect via ssh after your connection dropped (after some waiting time)? If this works, then you could try to start the upgrade in a screen session and check the state after a re-connect.
This is not a general solution for this issue but could give us a hint if we could improve the upgrade in the non NAT case with a meaningful logging (~"Your ssh session will be terminated - re-connect in a couple of minutes.") and working with `nohup` to keep the script running.

**Comment by Sujay-Radtonics** [comment]
@nstng , Since uplink_br0 looses it IP (which is the AGW mgmt IP), we cannot connect back to AGW using this IP. uplink_br0 will get back the IP and I can connect back after reboot of AGW. However, i see python3-ryu not upgraded, so upgrade has been partial. It may work if we run AGW via serial cable or console session. Have not got chance to try it yet

**Comment by panyogesh** [comment]
@Sujay-Radtonics : Is eth0 part of uplink_br0

**Comment by nstng** [comment]
> However, i see python3-ryu not upgraded, so upgrade has been partial.
Imho, the ryu upgrade should happen before the ovs upgrade. What upgrade script are you using - from master or from the 1.8 branch? The one from the 1.8 branch forces an update to the newest version of ryu.

(but this is independent to your connection loss problem)

**Comment by Sujay-Radtonics** [comment]
@nstng , I am using one from 1.8 branch. We had discussed about this earlier and you had put a fix for ryu upgrade

apt install -y magma -o Dpkg::Options::="--force-overwrite"

# update all direct dependencies of magma - this is needed for an update to 1.8 where
# ryu can be updated from an unpatched version 4.34 to a patched version 4.34-1.
apt install -y $(apt-cache depends magma | grep "Dep" | cut -d':' -f2)

My understanding is the OVS install is happening when apt install of magma is taking place, which is when we loose connectivity and probably script is aborted as well. And so ryu package is not updated.

**Comment by Sujay-Radtonics** [comment]
> @Sujay-Radtonics : Is eth0 part of uplink_br0

@panyogesh - Yes, that is how it works when we put AGW in non-NAT mode right ?
I am yet to try doing upgrade using console or the eth1 interface.

**Comment by nstng** [comment]
> My understanding is the OVS install is happening when apt install of magma is taking place, which is when we loose connectivity and probably script is aborted as well. And so ryu package is not updated.

Oh, ok - interesting, I was assuming your connection loss happens later in
```
#Upgrade OVS
ovs-kmod-upgrade.sh -y
```

**Comment by Sujay-Radtonics** [comment]
@panyogesh,  Looking forward to your kind suggestions on this. We need to finalize a procedure for customer installations

**Comment by shahrahman-fb** [comment]
Closing this issue as part of a repository cleanup effort. This issue has had no activity for over 2 years.

If this issue is still relevant, please reopen it or file a new issue with updated information. We are actively maintaining the project again and want to focus on current, actionable issues.

See: #15833 (Phase 1 cleanup), #15828 (tracking issue)
